ipa-core/src/ff/boolean_array.rs

```diff
@@ -156,9 +165,104 @@ macro_rules! boolean_array_impl_small {
     };
 }
 
-//macro for implementing Boolean array, only works for a byte size for which Block is defined
+/// Macro to implement `Serializable` trait for boolean arrays. Depending on the size, conversion from [u8; N] to `BAXX`
+/// can be fallible (if N is not a multiple of 8) or infallible. This macro takes care of it and provides the correct
+/// implementation. Because macros can't do math, a hint is required to advise it which implementation it should provide.
+#[rustfmt::skip]
+macro_rules! impl_serializable_trait {
+    ($name: ident, $bits: tt, fallible) => {
+        #[derive(thiserror::Error, Debug)]
+        #[error(
+            "The provided byte slice contains non-zero value(s) {0:?} in padding bits [{}..]",
+            $bits
+        )]
+        pub struct NonZeroPadding(Store);
+
+        /// The maximum value this boolean array can represent: `2^N - 1` where `N` is the number of bits.
+        /// Because of the [`restrictions`](https://docs.rs/bitvec/1.0.1/bitvec/array/struct.BitArray.html#usage),
+        /// for arrays with size `N` that is not a multiple of `8`, the `Store` capacity is greater than the
+        /// maximum value. For example: `BA20` (20 bit array) requires 3 bytes in memory and
+        /// store capacity (~16M) is greater than the maximum value for `BA20` which is (~1M).
+        const MAX_VALUE: Store = bitarr_init![1; $bits];
```

Comment:
Do you really need this?

To check if the value is in range, I think that you can do:

```rust
if raw_val[$bits..].not_any() {
  Ok(Self(raw_val))
} else {
  Err(NonZeroPadding(raw_val))
}
```